tests/integration/test_dq_rules.py

```diff
@@ -0,0 +1,153 @@
+            with working_directory(temp_dir):
+                logger.info(f"test_last_modified_in_dq_summary {gcp_application_credentials}")
+                target_table = f"{gcp_project_id}.{target_bq_result_dataset_name}.{target_bq_result_table_name}"
+                args = [
+                    "T1_DQ_1_VALUE_NOT_NULL,T2_DQ_1_EMAIL,T3_DQ_1_EMAIL_DUPLICATE",
+                    f"{temp_configs_dir}",
+                    f"--gcp_project_id={gcp_project_id}",
+                    f"--gcp_bq_dataset_id={gcp_bq_dataset}",
+                    f"--gcp_region_id={gcp_bq_region}",
+                    f"--target_bigquery_summary_table={target_table}",
+                    "--debug",
+                    "--summary_to_stdout",
+                    ]
+                result = runner.invoke(main, args)
+                print(result.output)
+                assert result.exit_code == 0
+
+                # Prepare dbt runtime
+                dbt_runner = DbtRunner(
+                    dbt_path=None,
+                    dbt_profiles_dir=None,
+                    environment_target="Dev",
+                    gcp_project_id=gcp_project_id,
+                    gcp_region_id=gcp_bq_region,
+                    gcp_bq_dataset_id=gcp_bq_dataset,
+                    gcp_service_account_key_path=gcp_sa_key,
+                    gcp_impersonation_credentials=gcp_impersonation_credentials,
+                )
+                dbt_path = dbt_runner.get_dbt_path()
+                invocation_id = get_dbt_invocation_id(dbt_path)
+                print(f"Dbt invocation id is {invocation_id}")
+                # Test the DQ expected results
+                try:
+                    sql = f"""SELECT rule_binding_id, rule_id, column_id,
+                    dimension, metadata_json_string, progress_watermark,
+                    rows_validated, complex_rule_validation_errors_count,
+                    success_count, failed_count, null_count
+                     FROM
+                     `{gcp_project_id}.{target_bq_result_dataset_name}.{target_bq_result_table_name}`
+                     WHERE invocation_id='{invocation_id}'
+                    EXCEPT DISTINCT
+                    SELECT rule_binding_id, rule_id, column_id,
+                    dimension, metadata_json_string, progress_watermark,
+                    rows_validated, complex_rule_validation_errors_count,
+                    success_count, failed_count, null_count
+                     FROM `{gcp_project_id}.{target_bq_result_dataset_name}.{target_bq_result_table_name}_expected`;
+                    """
+                    print(f"Dq rules validation query string is \n {sql}")
+                    query_job = client.execute_query(sql)
+                    results = query_job.result()
+                    logger.info("Query done")
+                    rows = list(results)
+                    print(f"Query execution returned {len(rows)} rows")
+                    assert len(rows) == 0
```

Comment:
does this also print out the error rows if len(rows) is > 0?

Reply:
we need the test to print out all failing rows in case the assertion fails

Reply:
> does this also print out the error rows if len(rows) is > 0?

No, for now it doesn't print out the error rows

Reply:
> we need the test to print out all failing rows in case the assertion fails

Will add it in the test.